src/components/TipControl.vue

```diff
@@ -0,0 +1,234 @@
+
+<template>
+  <a
+    v-if="USE_DEEP_LINKS"
+    :href="deepLink"
+    target="_blank"
+    class="tip__content"
+  >
+    <img :src="iconTip">
+    <ae-amount
+      :amount="tip.retip_amount_ae.toString()"
+      :round="2"
+      class="vertical-align-mid"
+    />
+    <fiat-value
+      :amount="tip.retip_amount_ae.toString()"
+      class="vertical-align-mid"
+    />
+  </a>
+  <div
+    v-else
+    class="tip-url__wrapper"
+    title="Total amount of retips"
+  >
+    <div
+      v-if="show"
+      class="overlay"
+      @click="toggleTip(false)"
+    />
+    <div
+      class="tip__container_wrapper"
+      @:click.stop
+    >
+      <div
+        class="retip__content"
+        :class="[{ active: show }]"
+        @click="toggleTip(!show)"
+      >
+        <img
+          class="tip__icon"
+          :src="iconTip"
+        >
+        <ae-amount
+          :amount="tip.retip_amount_ae"
+          :round="2"
+        />
+        <fiat-value :amount="tip.retip_amount_ae.toString()" />
+      </div>
+      <div
+        v-if="show"
+        class="tip__container"
+      >
+        <loading :show-loading="showLoading" />
+        <div
+          v-show="error && !showLoading"
+          class="text-center mb-2"
+        >
+          An error occurred while sending tip
+        </div>
+        <form
+          v-if="!showLoading"
+          @submit.prevent
```

Comment:
doesn't this work `@submit.prevent="submitAction()"` (and remove it from the button)? Enter on the focused input emits submit event to the form, so the form should handle it. The same way button=submit emits the same event to the form, so we usually handle the event there. Then other onclick stuff become useless (including the one in ae-button). If `<ae-button>` is to be used as generic button, so we better make type to be configurable (as component property)